src/OpenTelemetry/Logs/OpenTelemetryLogger.cs

```diff
@@ -64,14 +64,18 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 ref LogRecordData data = ref record.Data;
 
                 data.TimestampBacking = DateTime.UtcNow;
-                data.CategoryName = this.categoryName;
                 data.LogLevel = logLevel;
                 data.EventId = eventId;
                 data.Message = provider.IncludeFormattedMessage ? formatter?.Invoke(state, exception) : null;
                 data.Exception = exception;
 
                 LogRecordData.SetActivityContext(ref data, Activity.Current);
 
+                LogRecordAttributeList attributes = default;
+
+                // TODO: Confirm if this name for attribute is good.
+                attributes.Add("dotnet.ilogger.category", this.categoryName);
```

Comment:
What's the perf drop (from both writer and reader perspective)?

Reply:
Yea agreed gotta measure this. Before I spend time on that though, I would like to first get a sense whether folks agree that obsoleting CategoryName is the route we want to take.

Reply:
I put my quick thinking here https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491#issuecomment-1196039348.

Reply:
Is this doing anything? 🤣 Seems to be adding the attribute to the stack and then dropping it on the floor. If we wanted to drop CategoryName & EventId here we would have to turn on the buffering feature so we could add new tags to [AttributeStorage](https://github.com/open-telemetry/opentelemetry-dotnet/blob/f089ec75bdb3630ee3c77635727ea0c5c4144436/src/OpenTelemetry/Logs/LogRecord.cs#L35). Perf hit would be minor for users already doing buffering/batching but for users doing re-entrant processor to something like ETW (GenevaLogExporter) the perf hit would be significant!

I like @reyang's idea to just wait and see what else is coming and then tackle the weirdness. I would also love to drop LogRecord.State and only expose LogRecord.StateValues. Having the two is super confusing and it is kind of undefined what exporters should do with them?

Reply:
Can't say I feel all that bad about my application of [Cunningham's Law](https://meta.wikimedia.org/wiki/Cunningham%27s_Law) in this PR 😆. I am closing it because you've convinced me of the better option for now: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491#issuecomment-1197249067